addons/l10n_ro_edi/models/ciusro_document.py

```diff
@@ -0,0 +1,198 @@
+                    return self._handle_error(_('CIUS-RO XML not found'))
+                method = 'POST'
+                params = {'standard': 'UBL' if self.move_id.move_type == 'out_invoice' else 'CN',
+                          'cif': self.move_id.company_id.vat.replace('RO', '')}
+                data = xml_data
+            case 'stareMesaj':
+                method = 'GET'
+                params = {'id_incarcare': self.key_loading}
+                data = None
+            case 'descarcare':
+                method = 'GET'
+                params = {'id': self.key_download}
+                data = None
+            case _:
+                raise UserError(_('Invalid request endpoint "%s"') % endpoint)
+
+        try:
+            requester = session if session else requests
+            response = requester.request(method=method, url=url, params=params, data=data, headers=headers, timeout=10)
+        except requests.HTTPError as e:
+            return self._handle_error(str(e))
+        if response.status_code == 400:
+            error_json = response.json()
+            return self._handle_error(error_json['message'])
+        if response.status_code == 403:
+            return self._handle_error(_('Access token is forbidden'))
+
+        return response
+
+    def _request_ciusro_send_invoice(self, xml_data):
+        self.ensure_one()
+        response = self._make_request('upload', xml_data)
+        if not response:
+            return
+
+        root = etree.fromstring(response.content)
+        res_status = root.get('ExecutionStatus')
+        if res_status == '1':
+            error_elements = root.findall('.//ns:Errors', namespaces=NS_UPLOAD)
+            error_messages = [error_element.get('errorMessage') for error_element in error_elements]
+            return self._handle_error('\n'.join(error_messages))
+        else:
+            self.state = 'sending'
+            self.key_loading = root.get('index_incarcare')
+
+    def _request_ciusro_fetch_status(self, session=None):
+        self.ensure_one()
+        response = self._make_request('stareMesaj', session=session)
+        if not response:
+            return
+
+        root = etree.fromstring(response.content)
+        error_elements = root.findall('.//ns:Errors', namespaces=NS_STATUS)
+        if error_elements:
+            error_messages = [error_element.get('errorMessage') for error_element in error_elements]
+            return self._handle_error('\n'.join(error_messages))
+
+        state_status = root.get('stare')
+        self.key_download = root.get('id_descarcare')
+        if state_status in ('nok', 'ok'):
+            self.state = state_status
+
+    def _request_ciusro_download_answer(self):
+        self.ensure_one()
+        response = self._make_request('descarcare')
+        if not response:
+            return
+
+        # E-Factura gives download response in ZIP format
+        zip_ref = zipfile.ZipFile(io.BytesIO(response.content))
```

Comment:
What is in this zip? If it's only a signature plus an xml, what about unzipping and storing the 2 attachments on the document?